tests/sync/app.cpp

```diff
@@ -132,6 +132,24 @@ class IntTestTransport : public GenericNetworkTransport {
     }
 };
 
+class CustomErrorTransport : public GenericNetworkTransport {
+public:
+    CustomErrorTransport(int code, std::string message)
```

Comment:
[nit] `CustomErrorTransport(int code, const std::string& message) : m_code(code), m_message(message) {}`

We can also localise this transport to the test unless you think we will be re-using it.